kratos/python_scripts/analysis_stage.py

```diff
@@ -191,6 +191,28 @@ def ChangeMaterialProperties(self):
         """this function is where the user could change material parameters as a part of the solution step """
         pass
 
+    def ReInitializeSolver(self):
```

Comment:
this should definitely approved by @technicalcommittee

Reply:
I am really scared of this one :)

Reply:
There is no other option

Reply:
I don't think this is correct. For example imagine that you arrive to a given level of deformation. If you reinitialize the elements that initial deformation would be considered as the reference.